### PR TITLE
[SMALLFIX] Fix recursive flag for TTL Free

### PR DESCRIPTION
**Pull request by apc999**

### CODE REVIEW

File: core/server/src/main/java/alluxio/master/file/FileSystemMaster.java [hunk 1]
```diff
@@ -3101,7 +3101,7 @@ public void heartbeat() {
               LOG.debug("File {} is expired. Performing action {}", inode.getName(), ttlAction);
               switch (ttlAction) {
                 case FREE:
-                  free(path, FreeOptions.defaults().setForced(true));
+                  free(path, FreeOptions.defaults().setForced(true).setRecursive(true));
```
Comment (jsimsa) [inline core/server/src/main/java/alluxio/master/file/FileSystemMaster.java]: nit: would it make sense to change the `DELETE` case to also always use recursive == true for consistency?

Comment (apc999) [inline core/server/src/main/java/alluxio/master/file/FileSystemMaster.java]: done

**Comment by apc999** [comment]
@jsimsa I updated this PR after your LGTM. PTAL. thanks

File: core/server/src/main/java/alluxio/master/file/FileSystemMaster.java [hunk 2]
```diff
@@ -3098,10 +3098,16 @@ public void heartbeat() {
           if (path != null) {
             try {
               TtlAction ttlAction = inode.getTtlAction();
-              LOG.debug("File {} is expired. Performing action {}", inode.getName(), ttlAction);
+              LOG.debug("Path {} has expired. Performing action {}", path.getPath(), ttlAction);
```
Comment (jsimsa) [inline core/server/src/main/java/alluxio/master/file/FileSystemMaster.java]: nit: `Path {} TTL has expired. ...`

File: core/server/src/main/java/alluxio/master/file/FileSystemMaster.java [hunk 3]
```diff
@@ -3112,11 +3118,12 @@ public void heartbeat() {
...
                   if (inode.isDirectory()) {
                     delete(path, true);
+                  } else {
+                    delete(path, false);
                   }
-                  delete(path, false);
                   break;
                 default:
-                  LOG.error("Unknown TtlAction.");
+                  LOG.error("Unknown TtlAction {}", ttlAction);
```
Comment (jsimsa) [inline core/server/src/main/java/alluxio/master/file/FileSystemMaster.java]: nit: `Unknown TTL action {}`

File: core/server/src/main/java/alluxio/master/file/FileSystemMaster.java (hunk 2 above)
Comment (apc999) [inline core/server/src/main/java/alluxio/master/file/FileSystemMaster.java]: fixed

File: core/server/src/main/java/alluxio/master/file/FileSystemMaster.java (hunk 3 above)
Comment (apc999) [inline core/server/src/main/java/alluxio/master/file/FileSystemMaster.java]: done

**Comment by apc999** [comment]
@jsimsa  fixed. thanks

**Pull request merged**